### DISCUSSION TITLE
How to get all the layers list, which are editable

### DISCUSSION THREAD

**Discussion by Meet-Miyani** [Tencent/libpag#467] [Q&A] [answered]
There are methods :
   it.getLayersByName()
            it.getLayersByEditableIndex()
            it.getLayersUnderPoint()

But all this methods requires some parameter values to be passed in, I am not sure what to pass. Please Help

**Comment by lvpengwei** [comment, marked as answer]
You can use [numTexts](https://github.com/Tencent/libpag/blob/v4.0.5.17/android/libpag/src/main/java/org/libpag/PAGFile.java#L56) or [numImages](https://github.com/Tencent/libpag/blob/v4.0.5.17/android/libpag/src/main/java/org/libpag/PAGFile.java#L61) to get the number of replaceable layers, and then use [getLayersByEditableIndex](https://github.com/Tencent/libpag/blob/v4.0.5.17/android/libpag/src/main/java/org/libpag/PAGFile.java#L102) to get them one by one.

**Reply by Meet-Miyani** [reply]
Thanks, man. That seems a solution, and do you know how to get the Height and width of each Image layer ?

**Reply by lvpengwei** [reply]
Maybe you can use [getBounds](https://github.com/Tencent/libpag/blob/v4.0.5.17/android/libpag/src/main/java/org/libpag/PAGLayer.java#L161).
